sdk/src/utils/withExecAndTransfer.ts

```diff
@@ -39,14 +39,13 @@ type WithExecAndTransferParams = {
 export function withExecAndTransfer(
   params: WithExecAndTransferParams,
 ): Call<typeof middlemanABI> {
-  const { middleman } = useOmniContext()
   const { call, transfer } = params
   const _callData = encodeFunctionData({ ...call })
 
   return {
     ...call,
     abi: middlemanABI,
-    target: middleman,
+    target: params.middlemanAddress,
```

Comment:
thoughts on still using `useOmniContracts` here, so users don't need to pass in middleman address

Reply:
oh yeah it's a util function good call. keep it as is then yeah